src/notation/view/pianokeyboard/pianokeyboardcontroller.cpp

```diff
@@ -133,9 +133,9 @@ void PianoKeyboardController::updateNotesKeys(const std::vector<const Note*>& re
     };
 
     for (const mu::engraving::Note* note : receivedNotes) {
-        newKeys.insert(static_cast<piano_key_t>(note->epitch()));
+        newKeys.insert(static_cast<piano_key_t>(useWrittenPitch() ? note->epitch() : note->ppitch()));
```

Comment:
Perhaps make `useWrittenPitch` a local variable outside the `for` loop; then the `useWrittenPitch()` method is not necessary anymore. (Or, if you prefer keeping that method, better make it private.)